Sources/Skylab/SkylabUser.swift

```diff
@@ -7,61 +7,182 @@
+            self.user.deviceManufacturer = deviceManufacturer
+            return self
+        }
+
+        public func setDeviceModel(_ deviceModel: String?) -> Builder {
+            self.user.deviceModel = deviceModel
+            return self
+        }
+
+        public func setCarrier(_ carrier: String?) -> Builder {
+            self.user.carrier = carrier
+            return self
+        }
+
+        public func setLibrary(_ library: String?) -> Builder {
+            self.user.library = library
+            return self
+        }
+
+        public func setUserProperties(_ userProperties: [String: String]?) -> Builder {
+            self.user.userProperties = userProperties
+            return self
+        }
+
+        public func setUserProperty(_ property: String, value: String) -> Builder {
+            guard var userProperties = user.userProperties else {
+                self.user.userProperties = [property: value]
+                return self
+            }
+            userProperties[property] = value
+            return self
+        }
+
+        public func copyUser(_ user: SkylabUser) -> Builder {
+            if let userId = user.userId {
+                self.user.userId = userId
+            }
+            if let deviceId = user.deviceId {
+                self.user.deviceId = deviceId
+            }
+            if let country = user.country {
+                self.user.country = country
+            }
+            if let region = user.region {
+                self.user.region = region
+            }
+            if let city = user.city {
+                self.user.city = city
+            }
+            if let language = user.language {
+                self.user.language = language
+            }
+            if let platform = user.platform {
+                self.user.platform = platform
+            }
+            if let version = user.version {
+                self.user.version = version
+            }
+            if let os = user.os {
+                self.user.os = os
+            }
+            if let dma = user.dma {
+                self.user.dma = dma
+            }
+            if let deviceFamily = user.deviceFamily {
+                self.user.deviceFamily = deviceFamily
+            }
+            if let deviceType = user.deviceType {
+                self.user.deviceType = deviceType
+            }
+            if let deviceManufacturer = user.deviceManufacturer {
+                self.user.deviceManufacturer = deviceManufacturer
+            }
+            if let deviceModel = user.deviceModel {
+                self.user.deviceModel = deviceModel
+            }
+            if let carrier = user.carrier {
+                self.user.carrier = carrier
+            }
+            if let library = user.library {
+                self.user.library = library
+            }
+            if let userProperties = user.userProperties {
+                self.user.userProperties = userProperties
```

Comment:
Is this safe to do? Would this share userProperties object between the copies and produce the same bug I had introduced in Android? (Not sure how doing a copy of the userProperties object would affect Equatable though)

Reply:
Good question. Quick test:

```swift
var map: [String:String] = [:]
var map2 = map
map2["test"] = "test"
print(map)
print(map2)
```

output
```
[:]
["test": "test"]
```

so we should be in the clear